src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -441,6 +445,156 @@ def create(
+    """
+    Temporary Strategy
+    """
+
+    def __init__(
+        self,
+        max_batch_size: int,
+        max_batch_time: float,
+    ):
+        self.__max_batch_time = max_batch_time
+        self.__max_batch_size = max_batch_size
+
+    def create(
+        self, commit: Callable[[Mapping[Partition, Position]], None]
+    ) -> ProcessingStrategy[KafkaPayload]:
+        transform_step = TransformStep(next_step=SimpleProduceStep(commit))
+        strategy = BatchMessages(transform_step, self.__max_batch_time, self.__max_batch_size)
+        return strategy
+
+
+class TransformStep(ProcessingStep[MessageBatch]):
+    """
+    Temporary Transform Step
+    """
+
+    def __init__(
+        self,
+        next_step: ProcessingStep[MessageBatch],
+    ) -> None:
+        self.__process_messages = process_messages
+        self.__next_step = next_step
+        self.__closed = False
+        self.__metrics = get_metrics()
+
+    def poll(self) -> None:
+        self.__next_step.poll()
+
+    def submit(self, message: Message[MessageBatch]) -> None:
+        assert not self.__closed
+
+        with self.__metrics.timer("transform_step.process_messages"):
+            transformed_message = self.__process_messages(message)
+
+        self.__next_step.submit(
+            Message(
+                message.partition,
+                message.offset,
+                transformed_message,
+                message.timestamp,
+            )
+        )
+
+    def close(self) -> None:
+        self.__closed = True
+
+    def terminate(self) -> None:
+        self.__closed = True
+
+        logger.debug("Terminating %r...", self.__next_step)
+        self.__next_step.terminate()
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        self.__next_step.close()
+        self.__next_step.join(timeout)
+
+
+class SimpleProduceStep(ProcessingStep[MessageBatch]):  # type: ignore
+    """
+    Temporary Produce Step
+    """
+
+    def __init__(
+        self,
+        commit_function: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+        snuba_metrics_producer = Producer(
+            kafka_config.get_kafka_producer_cluster_options(snuba_metrics["cluster"]),
+        )
+        producer = snuba_metrics_producer
+        self.__producer = producer
+        self.__producer_topic = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS].get(
+            "topic", "snuba-metrics"
+        )
+        self.__commit_function = commit_function
+
+        self.__closed = False
+        self.__metrics = get_metrics()
+
+        self.__produced_messages: MutableMapping[Partition, OffsetRange] = {}
+
+    def poll(self) -> None:
+        if self.__produced_messages.keys():
+            self._commit(2.0)
+
+    def submit(self, outer_message: Message[MessageBatch]) -> None:
+        for message in outer_message.payload:
```

Comment:
I think it will be better to keep a different batch size between the batches you send to the transform step and the batch you use to commit.
Let's say you increase the outer message size to 100. 
This would still make you commit and flush every 100 messages. Which means 80 times a second if we can keep up with the 8000 messages per second.
That is still a lot.

While the outer message size should probably be between 10 and 100, this batch is more likely to be around 1000. So I would split the two.

Which means this step needs to accumulate messages. Thus I would still suggest to use the CollectStep since it does the accumulation for you.

Do like the Clickhouse one:
You provide these to the CollectStep:
- a strategy that processes one message (produces it) on submit. On close it vould flush
- the commit function.
The collectstep does the rest.